src/scheduler/Location.java

```diff
@@ -26,6 +26,22 @@ public enum Location {
         this.campusName = campusName;
     }
 
+    /**
+     * Checks if a given location string is valid by comparing it to the known building names
+     * in the Location enum.
+     *
+     * @param givenUserLocation The location string to validate.
+     * @return true if the location string is a valid building name, false otherwise.
+     */
+    public static boolean isValidLocation(String givenUserLocation) {
+        for (Location location : values()) {
+            if (location.buildingName.equalsIgnoreCase(givenUserLocation)) {
```

Comment:
You are comparing it to the full building name? That is the other source of your location bug.